sdk/messaging/azeventhubs/producer_client.go

```diff
@@ -109,43 +109,30 @@ type NewEventDataBatchOptions struct {
 // NewEventDataBatch can be used to create a batch that contain multiple events.
 // If the operation fails it can return an *azeventhubs.Error type if the failure is actionable.
 func (pc *ProducerClient) NewEventDataBatch(ctx context.Context, options *NewEventDataBatchOptions) (*EventDataBatch, error) {
-	if options == nil {
-		options = &NewEventDataBatchOptions{}
-	}
-
-	if options.PartitionID != nil && options.PartitionKey != nil {
-		return nil, errors.New("either PartitionID or PartitionKey can be set, but not both")
-	}
+	var batch *EventDataBatch
 
-	var batch EventDataBatch
+	partitionID := anyPartitionID
 
-	if options.PartitionID != nil {
-		// they want to send to a particular partition. The batch size should be the same for any
-		// link but we might as well use the one they're going to send to.
-		batch.partitionID = options.PartitionID
-	} else if options.PartitionKey != nil {
-		batch.partitionKey = options.PartitionKey
+	if options != nil && options.PartitionID != nil {
+		partitionID = *options.PartitionID
 	}
 
-	err := pc.links.Retry(ctx, exported.EventProducer, "NewEventDataBatch", getPartitionID(batch.partitionID), pc.retryOptions, func(ctx context.Context, lwid internal.LinkWithID[amqpwrap.AMQPSenderCloser]) error {
-		if options.MaxBytes == 0 {
-			batch.maxBytes = lwid.Link.MaxMessageSize()
-			return nil
-		}
+	err := pc.links.Retry(ctx, exported.EventProducer, "NewEventDataBatch", partitionID, pc.retryOptions, func(ctx context.Context, lwid internal.LinkWithID[amqpwrap.AMQPSenderCloser]) error {
+		tmpBatch, err := newEventDataBatch(lwid.Link, options)
```

Comment:
Why is this in a retry?  It doesn't appear to perform any IO.

Reply:
It doesn't, but it requires the link to be created so we can get the max message size.